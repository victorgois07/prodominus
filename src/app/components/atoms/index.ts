export * from "./Button";
export * from "./Error";
export * from "./Input";
export * from "./Label";
export * from "./Loading";
export * from "./Toast";
