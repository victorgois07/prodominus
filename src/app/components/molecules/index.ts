export * from "./CategoryFilter";
export * from "./ConfirmDialog";
export * from "./FormField";
export * from "./Modal";
export * from "./Pagination";
export * from "./SortFilter";
