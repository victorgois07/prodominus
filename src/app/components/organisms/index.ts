export * from "./FavoritesManagement";
export * from "./ProductCard";
export * from "./ProductFilters";
export * from "./ProductForm";
export * from "./ProductGrid";
export * from "./ProductManagement";
