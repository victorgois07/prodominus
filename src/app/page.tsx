"use client";

import { useProducts } from "@/business/hooks/useProducts";
import { useProductFilters } from "@/business/store/useProductFilters";
import {
  ConfirmDialog,
  Error,
  Loading,
  Pagination,
  ProductFilters,
  ProductManagement,
} from "./components";

export default function Home() {
  const { loading, error, totalPages } = useProducts();
  const { page, setPage } = useProductFilters();

  if (loading) return <Loading />;
  if (error) return <Error message={error} />;

  return (
    <main className="container mx-auto px-4 py-8">
      <h1 className="text-3xl font-bold text-gray-900 mb-8">Products</h1>

      <ProductFilters />

      <div className="mb-6 flex flex-col gap-4">
        <ProductManagement />
      </div>

      <Pagination page={page} totalPages={totalPages} onPage={setPage} />
      <ConfirmDialog />
    </main>
  );
}
