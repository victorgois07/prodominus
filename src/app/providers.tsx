"use client";

import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { useState } from "react";

export function Providers({ children }: { children: React.ReactNode }) {
  const [queryClient] = useState(
    () =>
      new QueryClient({
        defaultOptions: {
          queries: {
            staleTime: 1000 * 60 * 5,
            retry: 1,
            refetchOnWindowFocus: false,
            retryOnMount: true,
            refetchOnReconnect: true,
          },
        },
      })
  );

  return (
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  );
}
