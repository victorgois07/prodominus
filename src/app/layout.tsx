import type { Metadata } from "next";
import { Inter } from "next/font/google";
import { LoadingBackdrop } from "./components/atoms/LoadingBackdrop";
import "./globals.css";
import { Providers } from "./providers";

const inter = Inter({ subsets: ["latin"] });

export const metadata: Metadata = {
  title: "Prodominus - Product Management",
  description: "A modern product management system built with Next.js",
};

export default function RootLayout({
  children,
}: Readonly<{ children: React.ReactNode }>) {
  return (
    <html lang="en">
      <body className={`${inter.className} antialiased`}>
        <Providers>
          <header className="bg-white shadow-sm">
            <div className="container mx-auto px-4 py-4">
              <h1 className="text-2xl font-bold text-balance text-gray-900">
                Prodominus
              </h1>
            </div>
          </header>

          <main>{children}</main>

          <footer className="bg-gray-50 mt-8">
            <div className="container mx-auto px-4 py-6">
              <p className="text-center text-gray-600 text-sm">
                Â© {new Date().getFullYear()} Prodominus. All rights reserved.
              </p>
            </div>
          </footer>

          <LoadingBackdrop />
        </Providers>
      </body>
    </html>
  );
}
