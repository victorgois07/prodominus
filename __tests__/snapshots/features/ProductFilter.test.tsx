import { ProductFilters } from "@/app/components";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { render } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";

vi.mock("@/business/hooks/useProducts", () => ({
  useProducts: () => ({
    categories: ["electronics", "clothing", "books", "home", "sports"],
    category: "",
    setCategory: () => {},
    sort: "asc",
    setSort: () => {},
    page: 1,
    setPage: () => {},
    totalPages: 1,
    products: [],
    loading: false,
    error: null,
  }),
}));

describe("ProductFilter Snapshot", () => {
  const mockCategories = ["electronics", "clothing", "books", "home", "sports"];
  const queryClient = new QueryClient();

  it("should match snapshot with categories", () => {
    const { container } = render(
      <QueryClientProvider client={queryClient}>
        <ProductFilters categories={mockCategories} />
      </QueryClientProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
