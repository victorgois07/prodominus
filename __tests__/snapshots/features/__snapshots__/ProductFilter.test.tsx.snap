// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductFilter Snapshot > should match snapshot with categories 1`] = `
<div>
  <div
    class="flex flex-col md:flex-row gap-4 mb-6"
  >
    <input
      class="shadow border rounded focus:shadow-outline mb-6 appearance-none bg-transparent border-none w-full text-gray-700 mr-3 py-1 px-2 leading-tight focus:outline-none"
      placeholder="Search products"
      type="text"
      value=""
    />
    <div
      class="mb-6"
    >
      <label
        class="block mb-1 font-medium"
        for="category-filter"
      >
        Categoria
      </label>
      <select
        class="block w-full p-2 border border-gray-300 rounded"
        id="category-filter"
      >
        <option
          value=""
        >
          Todas
        </option>
        <option
          value="electronics"
        >
          electronics
        </option>
        <option
          value="clothing"
        >
          clothing
        </option>
        <option
          value="books"
        >
          books
        </option>
        <option
          value="home"
        >
          home
        </option>
        <option
          value="sports"
        >
          sports
        </option>
      </select>
    </div>
    <div
      class="mb-6"
    >
      <label
        class="block mb-1 font-medium"
        for="sort-filter"
      >
        Ordenar por
      </label>
      <select
        class="block w-full p-2 border border-gray-300 rounded"
        id="sort-filter"
      >
        <option
          value=""
        >
          Padrão
        </option>
        <option
          value="asc"
        >
          Preço: Menor para Maior
        </option>
        <option
          value="desc"
        >
          Preço: Maior para Menor
        </option>
      </select>
    </div>
  </div>
</div>
`;
